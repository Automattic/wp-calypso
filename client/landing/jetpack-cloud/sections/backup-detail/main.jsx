/**
 * External dependencies
 */
import React, { Component } from 'react';
import { connect } from 'react-redux';

/**
 * Internal dependencies
 */
import { getSelectedSiteId } from 'state/ui/selectors';

class JetpackCloudBackupDetailPage extends Component {
	render() {
		const { siteId, backupId } = this.props;

		return (
			<div>
				<div>Welcome to the backup detail page</div>
				<div>Site ID: { siteId }</div>
				<div>Backup ID: { backupId }</div>
			</div>
		);
	}
}

export default connect( state => {
	const siteId = getSelectedSiteId( state );

	return {
		siteId,
	};
} )( JetpackCloudBackupDetailPage );
