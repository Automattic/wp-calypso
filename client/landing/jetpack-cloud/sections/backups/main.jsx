/**
 * External dependencies
 */
import { connect } from 'react-redux';
import { isMobile } from '@automattic/viewport';
import page from 'page';
import React, { Component } from 'react';

/**
 * Internal dependencies
 */
import DocumentHead from 'components/data/document-head';
import { updateFilter } from 'state/activity-log/actions';
import { getBackupAttemptsForDate, getDailyBackupDeltas, getEventsInDailyBackup } from './utils';
import { getSelectedSiteId } from 'state/ui/selectors';
import { requestActivityLogs } from 'state/data-getters';
import { withLocalizedMoment } from 'components/localized-moment';
import BackupDelta from '../../components/backup-delta';
import DailyBackupStatus from '../../components/daily-backup-status';
import DatePicker from '../../components/date-picker';
import getRewindState from 'state/selectors/get-rewind-state';
import getSelectedSiteSlug from 'state/ui/selectors/get-selected-site-slug';
import QueryRewindState from 'components/data/query-rewind-state';
import QuerySitePurchases from 'components/data/query-site-purchases';
import Main from 'components/main';
import SidebarNavigation from 'my-sites/sidebar-navigation';
import getActivityLogFilter from 'state/selectors/get-activity-log-filter';
import Filterbar from 'my-sites/activity/filterbar';
import ActivityCard from '../../components/activity-card';
import siteSupportsRealtimeBackup from 'state/selectors/site-supports-realtime-backup';
import Pagination from 'components/pagination';

/**
 * Style dependencies
 */
import './style.scss';

const PAGE_SIZE = 10;

class BackupsPage extends Component {
	constructor( props ) {
		super( props );

		this.state = {
			selectedDate: new Date(),
		};
	}

	onDateChange = date => {
		this.setState( { selectedDate: date } );
	};

	onDateRangeSelection = () => {
		//todo: go to the log activity view
	};

	isEmptyFilter = filter => {
		if ( ! filter ) {
			return true;
		}
		if ( filter.group || filter.on || filter.before || filter.after ) {
			return false;
		}
		if ( filter.page !== 1 ) {
			return false;
		}
		return true;
	};

	TO_REMOVE_getSelectedDateString = () => {
		const { moment } = this.props;
		const { selectedDate } = this.state;
		return moment.parseZone( selectedDate ).toISOString( true );
	};

	renderMain() {
		const { allowRestore, hasRealtimeBackups, logs, moment, siteId, siteSlug } = this.props;
		const { selectedDate } = this.state;
		const selectedDateString = this.TO_REMOVE_getSelectedDateString();

		const backupAttempts = getBackupAttemptsForDate( logs, selectedDateString );
		const deltas = getDailyBackupDeltas( logs, selectedDateString );
		const realtimeEvents = getEventsInDailyBackup( logs, selectedDateString );

		return (
			<Main>
				<DocumentHead title="Backups" />
				<SidebarNavigation />
				<QueryRewindState siteId={ siteId } />
				<QuerySitePurchases siteId={ siteId } />

				<DatePicker
					onDateChange={ this.onDateChange }
					onDateRangeSelection={ this.onDateRangeSelection }
					selectedDate={ selectedDate }
					siteId={ siteId }
				/>
				<DailyBackupStatus
					allowRestore={ allowRestore }
					date={ selectedDateString }
					backupAttempts={ backupAttempts }
					siteSlug={ siteSlug }
				/>
				<BackupDelta
					{ ...{
						deltas,
						backupAttempts,
						hasRealtimeBackups,
						realtimeEvents,
						allowRestore,
						moment,
						siteSlug,
					} }
				/>
			</Main>
		);
	}

	changePage = pageNumber => {
		this.props.selectPage( this.props.siteId, pageNumber );
		window.scrollTo( 0, 0 );
	};

	renderActivityLog() {
		const { allowRestore, filter, logs, moment, siteId } = this.props;
		const { page: requestedPage } = filter;

		const actualPage = Math.max(
			1,
			Math.min( requestedPage, Math.ceil( logs.length / PAGE_SIZE ) )
		);
		const theseLogs = logs.slice( ( actualPage - 1 ) * PAGE_SIZE, actualPage * PAGE_SIZE );

		const cards = theseLogs.map( activity => (
			<ActivityCard
				{ ...{
					key: activity.activityId,
					moment,
					activity,
					allowRestore,
				} }
			/>
		) );

		return (
			<div>
				<div>Find a backup or restore point</div>
				<div>
					This is the complete event history for your site. Filter by date range and/ or activity
					type.
				</div>
				<Filterbar
					{ ...{
						siteId,
						filter,
						isLoading: false,
						isVisible: true,
					} }
				/>
				<Pagination
					compact={ isMobile() }
					className="backups__pagination"
					key="backups__pagination-top"
					nextLabel={ 'Older' }
					page={ actualPage }
					pageClick={ this.changePage }
					perPage={ PAGE_SIZE }
					prevLabel={ 'Newer' }
					total={ logs.length }
				/>
				{ cards }
				<Pagination
					compact={ isMobile() }
					className="backups__pagination"
					key="backups__pagination-bottom"
					nextLabel={ 'Older' }
					page={ actualPage }
					pageClick={ this.changePage }
					perPage={ PAGE_SIZE }
					prevLabel={ 'Newer' }
					total={ logs.length }
				/>
			</div>
		);
	}

	render() {
		const { filter } = this.props;

		return (
			<div className="backups__page">
				{ ! this.isEmptyFilter( filter ) ? this.renderActivityLog() : this.renderMain() }
			</div>
		);
	}
}

const mapStateToProps = state => {
	const siteId = getSelectedSiteId( state );

	//The section require a valid site, if not, redirect to backups
	if ( false === !! siteId ) {
		return page.redirect( '/backups' );
	}

	const filter = getActivityLogFilter( state, siteId );
	const logs = siteId && requestActivityLogs( siteId, filter );
	const rewind = getRewindState( state, siteId );
	const restoreStatus = rewind.rewind && rewind.rewind.status;
	const allowRestore =
		'active' === rewind.state && ! ( 'queued' === restoreStatus || 'running' === restoreStatus );

	return {
		allowRestore,
		filter,
		hasRealtimeBackups: siteSupportsRealtimeBackup( state, siteId ),
		logs: logs?.data ?? [],
		rewind,
		siteId,
		siteSlug: getSelectedSiteSlug( state ),
	};
};

const mapDispatchToProps = dispatch => ( {
	selectPage: ( siteId, pageNumber ) => dispatch( updateFilter( siteId, { page: pageNumber } ) ),
} );

export default connect( mapStateToProps, mapDispatchToProps )( withLocalizedMoment( BackupsPage ) );
