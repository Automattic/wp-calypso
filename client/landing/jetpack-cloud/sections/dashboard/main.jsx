/**
 * External dependencies
 */
import React, { Component } from 'react';
import { connect } from 'react-redux';

/**
 * Internal dependencies
 */
import { getCurrentUserName } from 'state/current-user/selectors';

class JetpackCloudDashboardPage extends Component {
	render() {
		return ( <div>Welcome to the dashboard, { this.props.currentUserName }!</div> );
	}
}

export default connect( state => {
	return {
		currentUserName: getCurrentUserName( state ),
	};
} )( JetpackCloudDashboardPage );
