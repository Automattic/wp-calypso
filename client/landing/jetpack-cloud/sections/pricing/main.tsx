/**
 * External dependencies
 */
import React from 'react';
import { useTranslate } from 'i18n-calypso';

/**
 * Internal dependencies
 */
import DocumentHead from 'components/data/document-head';
import Main from 'components/main';
import PageViewTracker from 'lib/analytics/page-view-tracker';

/**
 * Style dependencies
 */
import './style.scss';

const PricingPage = () => {
	const translate = useTranslate();

	return (
		<Main className="pricing">
			<DocumentHead title={ translate( 'Pricing' ) } />
			<PageViewTracker path="/pricing" title="Pricing" />

			<div className="pricing__title">
				<h2>{ translate( 'Welcome to the Pricing page' ) }</h2>
			</div>
		</Main>
	);
};

export default PricingPage;
