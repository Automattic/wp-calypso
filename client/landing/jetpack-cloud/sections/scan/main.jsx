/**
 * External dependencies
 */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Button } from '@automattic/components';
import { numberFormat, translate } from 'i18n-calypso';

/**
 * Internal dependencies
 */
import { getSelectedSite, getSelectedSiteSlug } from 'state/ui/selectors';
import { withLocalizedMoment } from 'components/localized-moment';
import SecurityIcon from 'landing/jetpack-cloud/components/security-icon';
import StatsFooter from 'landing/jetpack-cloud/components/stats-footer';
import ThreatItem from '../../components/threat-item';
import { isEnabled } from 'config';

import './style.scss';

class ScanPage extends Component {
	renderScanOkay() {
		const { siteSlug, moment, lastScanTimestamp } = this.props;

		return (
			<>
				<SecurityIcon className="scan__icon" />
				<h1 className="scan__header scan__header--okay">
					{ translate( 'Don’t worry about a thing' ) }
				</h1>
				<p>
					The last Jetpack scan ran <strong>{ moment( lastScanTimestamp ).fromNow() }</strong> and
					everything looked great.
				</p>
				{ isEnabled( 'jetpack-cloud/on-demand-scan' ) && (
					<Button
						primary
						href={ `/scan/${ siteSlug }/?scan-state=scanning` }
						className="scan__button"
					>
						{ translate( 'Scan now' ) }
					</Button>
				) }
			</>
		);
	}

	renderScanning() {
		return <p>Scanning!</p>;
	}

	renderThreats() {
		const { threats, site } = this.props;

		return (
			<>
				<h1 className="scan__header">{ translate( 'Your site may be at risk' ) }</h1>
				<p>
					{ translate(
						'The scan found {{strong}}%(threatCount)s{{/strong}} potential threat with {{strong}}%(siteName)s{{/strong}}.',
						'The scan found {{strong}}%(threatCount)s{{/strong}} potential threats with {{strong}}%(siteName)s{{/strong}}.',
						{
							args: {
								siteName: site.name,
								threatCount: numberFormat( threats.length ),
							},
							components: { strong: <strong /> },
							comment:
								'%(threatCount)s represents the number of threats currently identified on the site, and $(siteName)s is the name of the site.',
							count: threats.length,
						}
					) }
					<br />
					{ translate(
						'Please review them below and take action. We are {{a}}here to help{{/a}} if you need us.',
						{
							components: { a: <a href="https://jetpack.com/contact-support/" /> },
							comment: 'The {{a}} tag is a link that goes to a contact support page.',
						}
					) }
				</p>
				<div className="scan__threats">
					{ threats.map( threat => (
						<ThreatItem key={ threat.id } threat={ threat } />
					) ) }
				</div>
			</>
		);
	}

	renderScanError() {
		return <p>There is an error with the scan.</p>;
	}

	renderScanState() {
		switch ( this.props.scanState ) {
			case 'okay':
				return this.renderScanOkay();
			case 'scanning':
				return this.renderScanning();
			case 'threats':
				return this.renderThreats();
			case 'error':
				return this.renderScanError();
		}
	}

	render() {
		return (
			<div>
				{ this.renderScanState() }
				<StatsFooter
					header="Scan Summary"
					stats={ [
						{ name: 'Files', number: 1201 },
						{ name: 'Plugins', number: 4 },
						{ name: 'Themes', number: 3 },
					] }
					noticeText="Failing to plan is planning to fail. Regular backups ensure that should the worst happen, you are prepared. Jetpack Backups has you covered."
					noticeLink="https://jetpack/upgrade/backups"
				/>
			</div>
		);
	}
}

export default connect( state => {
	const site = getSelectedSite( state );
	const siteSlug = getSelectedSiteSlug( state );

	// TODO: Get state from actual API.
	const params = new URL( document.location ).searchParams;
	const scanState = params.get( 'scan-state' ) || 'okay';

	// TODO: Get threats from actual API.
	const threats = [
		{
			id: 1,
			title: 'Infected core file: index.php',
			action: null,
			detectionDate: '23 September, 2019',
			actionDate: null,
			description: {
				title: 'Unexpected string was found in: /htdocs/wp-admin/index.php',
				problem:
					'Jetpack has detected code that is often used in web-based "shell" programs. If you believe the file(s) have been infected they need to be cleaned.',
				fix:
					'To fix this threat, Jetpack will be deleting the file, since it’s not a part of the original WordPress.',
				details: 'This threat was found in the file: /htdocs/index.php',
			},
		},
		{
			id: 2,
			title: 'Infected Plugin: Contact Form 7',
			action: null,
			detectionDate: '17 September, 2019',
			actionDate: null,
			description: {
				title:
					'Unexpected file baaaaaad--file.php contains malicious code and is not part of the Plugin',
				problem:
					'Jetpack has detected code that is often used in web-based "shell" programs. If you believe the file(s) have been infected they need to be cleaned.',
				fix:
					'To fix this threat, Jetpack will be deleting the file, since it’s not a part of the original WordPress.',
				details: 'This threat was found in the file: /htdocs/sx--a4bp.php',
			},
		},
	];

	const lastScanTimestamp = Date.now() - 5700000; // 1h 35m.
	const nextScanTimestamp = Date.now() + 5700000;

	return {
		site,
		siteSlug,
		scanState,
		lastScanTimestamp,
		nextScanTimestamp,
		threats,
	};
} )( withLocalizedMoment( ScanPage ) );
