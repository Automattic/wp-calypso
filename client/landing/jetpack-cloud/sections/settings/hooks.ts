/**
 * External dependencies
 */
import { useSelector } from 'react-redux';

/**
 * Internal dependencies
 */
import getRewindState from 'state/selectors/get-rewind-state';
import { getSelectedSiteId } from 'state/ui/selectors';

export function useSelectedSiteId() {
	return useSelector( getSelectedSiteId );
}

export function useRewindState( siteId: number | string | null ) {
	return useSelector( ( state ) => getRewindState( state, siteId ) );
}
