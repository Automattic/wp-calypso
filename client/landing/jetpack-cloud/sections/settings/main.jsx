/**
 * External dependencies
 */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { localize } from 'i18n-calypso';

/**
 * Internal dependencies
 */
import DocumentHead from 'components/data/document-head';
import { getSelectedSiteId } from 'state/ui/selectors';
import RewindCredentialsForm from 'components/rewind-credentials-form';
import { Card } from '@automattic/components';
import getRewindState from 'state/selectors/get-rewind-state';
import QueryRewindState from 'components/data/query-rewind-state';
import Gridicon from 'components/gridicon';
import Main from 'components/main';
import SidebarNavigation from 'my-sites/sidebar-navigation';

/**
 * Style dependencies
 */
import './style.scss';

class SettingsPage extends Component {
	render() {
		const { rewind, siteId, translate } = this.props;

		const isConnected = 'active' === rewind.state;

		return (
			<Main className="settings">
				<DocumentHead title={ translate( 'Settings' ) } />
				<SidebarNavigation />
				<QueryRewindState siteId={ siteId } />
				<div className="settings__page-title">{ translate( 'Server connection details' ) }</div>
				{ isConnected && (
					<Card compact={ true } className="settings__connected">
						<Gridicon icon="checkmark-circle" />
						<div className="settings__details">
							<div className="settings__details-head">
								{ translate( 'Server status: Connected' ) }
							</div>
							<div>{ translate( 'One-click restores are enabled.' ) }</div>
						</div>
					</Card>
				) }
				{ ! isConnected && (
					<Card compact={ true } className="settings__disconnected">
						<Gridicon icon="cross-circle" />
						<div className="settings__details">
							<div className="settings__details-head">
								{ translate( 'Server status: Not connected' ) }
							</div>
							<div>
								{ translate(
									'Enter your server credentials to enable one-click restores for Backups. Find your server credentials.'
								) }
							</div>
						</div>
					</Card>
				) }
				<RewindCredentialsForm
					{ ...{
						allowCancel: false,
						role: 'main',
						siteId,
						showNotices: false,
					} }
				/>
			</Main>
		);
	}
}

export default connect( state => {
	const siteId = getSelectedSiteId( state );
	const rewind = getRewindState( state, siteId );

	return {
		siteId,
		rewind,
	};
} )( localize( SettingsPage ) );
