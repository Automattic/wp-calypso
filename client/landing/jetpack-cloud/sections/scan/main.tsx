/**
 * External dependencies
 */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Button, ProgressBar } from '@automattic/components';
import { translate } from 'i18n-calypso';
import { flowRight as compose } from 'lodash';

/**
 * Internal dependencies
 */
import DocumentHead from 'components/data/document-head';
import QueryJetpackScan from 'components/data/query-jetpack-scan';
import SecurityIcon from 'landing/jetpack-cloud/components/security-icon';
import ScanPlaceholder from 'landing/jetpack-cloud/components/scan-placeholder';
import StatsFooter from 'landing/jetpack-cloud/components/stats-footer';
import ScanThreats from 'landing/jetpack-cloud/components/scan-threats';
import { Scan, Site } from 'landing/jetpack-cloud/sections/scan/types';
import { isEnabled } from 'config';
import Gridicon from 'components/gridicon';
import Main from 'components/main';
import SidebarNavigation from 'my-sites/sidebar-navigation';
import PageViewTracker from 'lib/analytics/page-view-tracker';
import { getSelectedSite, getSelectedSiteId, getSelectedSiteSlug } from 'state/ui/selectors';
import getSiteUrl from 'state/sites/selectors/get-site-url';
import getSiteScanProgress from 'state/selectors/get-site-scan-progress';
import getSiteScanIsInitial from 'state/selectors/get-site-scan-is-initial';
import getSiteScanState from 'state/selectors/get-site-scan-state';
import { withLocalizedMoment } from 'components/localized-moment';
import contactSupportUrl from 'landing/jetpack-cloud/lib/contact-support-url';
import { recordTracksEvent } from 'state/analytics/actions';
import { triggerScanRun } from 'landing/jetpack-cloud/lib/trigger-scan-run';
import {
	withApplySiteOffset,
	applySiteOffsetType,
} from 'landing/jetpack-cloud/components/site-offset';

/**
 * Style dependencies
 */
import './style.scss';

interface Props {
	site: Site | null;
	siteId: number | null;
	siteSlug: string | null;
	siteUrl?: string;
	scanState?: Scan;
	scanProgress?: number;
	isInitialScan?: boolean;
	timezone: string | null;
	gmtOffset: number | null;
	moment: {
		utc: Function;
	};
	applySiteOffset: applySiteOffsetType;
	dispatchRecordTracksEvent: Function;
	dispatchScanRun: Function;
}

class ScanPage extends Component< Props > {
	renderProvisioning() {
		return (
			<>
				<SecurityIcon icon="in-progress" />
				{ this.renderHeader( translate( 'Preparing to scan' ) ) }
				<p>
					Lorem ipsum. We need to change this text. The scan was unable to process the themes
					directory and did not completed successfully. In order to complete the scan you will need
					to speak to support who can help determine what went wrong.
				</p>
				{ this.renderContactSupportButton() }
			</>
		);
	}

	renderHeader( text: i18nCalypso.TranslateResult ) {
		return <h1 className="scan__header">{ text }</h1>;
	}

	renderContactSupportButton() {
		const { dispatchRecordTracksEvent, siteUrl, siteId, scanState } = this.props;
		const scanStateType = scanState?.state;

		return (
			<Button
				primary
				target="_blank"
				rel="noopener noreferrer"
				href={ contactSupportUrl( siteUrl, scanStateType ) }
				className="scan__button"
				onClick={ () =>
					dispatchRecordTracksEvent( 'calypso_scan_error_contact_support', {
						scan_state: scanStateType,
						site_id: siteId,
					} )
				}
			>
				{ translate( 'Contact Support {{externalIcon/}}', {
					components: { externalIcon: <Gridicon icon="external" size={ 24 } /> },
				} ) }
			</Button>
		);
	}
	// @todo: missing copy and design for this state
	renderUnavailable() {
		return (
			<>
				<SecurityIcon icon="scan-error" />
				{ this.renderHeader( 'Scan is unavailable' ) }
				<p>
					{ translate(
						'The scan was unable to process the themes directory and did not completed ' +
							'successfully. In order to complete the scan you will need to speak to support ' +
							'who can help determine what went wrong.'
					) }
				</p>
				{ this.renderContactSupportButton() }
			</>
		);
	}

	renderScanOkay() {
		const { scanState, siteId, siteSlug, moment, dispatchScanRun, applySiteOffset } = this.props;
		const lastScanTimestamp = scanState?.mostRecent?.timestamp;
		let lastScanSiteTime = '';
		if ( lastScanTimestamp && applySiteOffset ) {
			lastScanSiteTime = applySiteOffset( moment.utc( lastScanTimestamp ) )?.fromNow();
		}

		return (
			<>
				<SecurityIcon />
				{ this.renderHeader( translate( 'Don’t worry about a thing' ) ) }
				<p>
					{ translate(
						/* translators: %s is a time string relative to now */
						'The last Jetpack scan ran {{strong}}%s{{/strong}} and everything ' +
							'looked great.' +
							'{{br/}}' +
							'Run a manual scan now or wait for Jetpack to scan your site later today.',
						{
							args: [ lastScanSiteTime ],
							components: {
								strong: <strong />,
								br: <br />,
							},
						}
					) }
				</p>
				{ isEnabled( 'jetpack-cloud/on-demand-scan' ) && (
					<Button
						primary
						href={ `/scan/${ siteSlug }` }
						className="scan__button"
						onClick={ () => dispatchScanRun( siteId ) }
					>
						{ translate( 'Scan now' ) }
					</Button>
				) }
			</>
		);
	}

	renderScanning() {
		const { scanProgress = 0, isInitialScan } = this.props;

		const heading =
			scanProgress === 0 ? translate( 'Preparing to scan' ) : translate( 'Scanning files' );

		return (
			<>
				<SecurityIcon icon="in-progress" />
				{ this.renderHeader( heading ) }
				<ProgressBar value={ scanProgress } total={ 100 } color="#069E08" />
				{ isInitialScan && (
					<p>
						{ translate(
							'Welcome to Jetpack Scan, we are taking a first look at your site now ' +
								'and the results will be with you soon.'
						) }
					</p>
				) }
				<p>
					{ translate(
						'We will send you an email once the scan completes, in the meantime feel ' +
							'free to continue to use your site as normal, you can check back on ' +
							'progress at any time.'
					) }
				</p>
			</>
		);
	}

	renderScanError() {
		const { siteId, siteSlug, dispatchScanRun } = this.props;

		return (
			<>
				<SecurityIcon icon="scan-error" />
				{ this.renderHeader( translate( 'Something went wrong' ) ) }
				<p>
					{ translate(
						"Jetpack Scan couldn't complete a scan of your site. Please check to see if your site is down " +
							"– if it's not, try again. If it is, or if Jetpack Scan is still having problems, contact our support team."
					) }
				</p>
				{ this.renderContactSupportButton() }
				{ isEnabled( 'jetpack-cloud/on-demand-scan' ) && (
					<Button
						href={ `/scan/${ siteSlug }` }
						className="scan__button scan__retry-bottom"
						onClick={ () => dispatchScanRun( siteId ) }
					>
						{ translate( 'Retry Scan' ) }
					</Button>
				) }
			</>
		);
	}

	renderScanState() {
		const { site, scanState } = this.props;
		if ( ! scanState || ! site ) {
			return <ScanPlaceholder />;
		}

		const { state, mostRecent, threats } = scanState;

		if ( state === 'provisioning' ) {
			return this.renderProvisioning();
		}

		// @todo: missing copy and design for these states
		if ( state === 'unavailable' ) {
			return this.renderUnavailable();
		}

		if ( state === 'scanning' ) {
			return this.renderScanning();
		}

		const errorFound = !! mostRecent?.error;
		const threatsFound = threats.length > 0;

		if ( errorFound && ! threatsFound ) {
			return this.renderScanError();
		}

		if ( threatsFound ) {
			return <ScanThreats threats={ threats } error={ errorFound } site={ site } />;
		}

		return this.renderScanOkay();
	}

	render() {
		const { siteId } = this.props;
		if ( ! siteId ) {
			return;
		}
		return (
			<Main className="scan__main">
				<DocumentHead title="Scanner" />
				<SidebarNavigation />
				<QueryJetpackScan siteId={ siteId } />
				<PageViewTracker path="/scan/:site" title="Scanner" />
				<div className="scan__content">{ this.renderScanState() }</div>
				<StatsFooter
					header="Scan Summary"
					noticeText={ translate(
						'Failing to plan is planning to fail. Regular backups ensure that should ' +
							'the worst happen, you are prepared. Jetpack Backup has you covered.'
					).toString() }
					noticeLink="https://jetpack.com/upgrade/backup"
				/>
			</Main>
		);
	}
}

export default connect(
	( state ) => {
		const site = getSelectedSite( state ) as Site;
		const siteId = getSelectedSiteId( state );
		const siteSlug = getSelectedSiteSlug( state );
		if ( ! siteId ) {
			return {
				site,
				siteId,
				siteSlug,
			};
		}
		const siteUrl = getSiteUrl( state, siteId ) ?? undefined;
		const scanState = ( getSiteScanState( state, siteId ) as Scan ) ?? undefined;
		const scanProgress = getSiteScanProgress( state, siteId ) ?? undefined;
		const isInitialScan = getSiteScanIsInitial( state, siteId );

		return {
			site,
			siteId,
			siteUrl,
			siteSlug,
			scanState,
			scanProgress,
			isInitialScan,
		};
	},
	{
		dispatchRecordTracksEvent: recordTracksEvent,
		dispatchScanRun: triggerScanRun,
	}
)( compose( withLocalizedMoment, withApplySiteOffset )( ScanPage ) );
