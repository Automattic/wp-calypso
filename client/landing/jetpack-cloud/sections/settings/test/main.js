/**
 * External dependencies
 */
import { shallow } from 'enzyme';
import React from 'react';

/**
 * Internal dependencies
 */
import * as hooks from '../hooks';
import { SettingsPage, ConnectionStatus } from '../main';

/**
 * Mocks
 */
jest.mock( '../hooks' );

describe( 'SettingsPage', () => {
	beforeEach( () => {
		jest.spyOn( hooks, 'useSelectedSiteId' ).mockReturnValue( 1 );
	} );

	afterAll( () => {
		jest.restoreAllMocks();
	} );

	test( 'should render as expected', () => {
		jest.spyOn( hooks, 'useRewindState' ).mockReturnValue( { state: 'active' } );

		const wrapper = shallow( <SettingsPage /> );

		expect( wrapper ).toMatchSnapshot();
	} );

	test( 'should render as expected when not connected', () => {
		jest.spyOn( hooks, 'useRewindState' ).mockReturnValue( { state: 'inactive' } );

		const wrapper = shallow( <SettingsPage /> );

		expect( wrapper ).toMatchSnapshot();
	} );

	test( 'should render as expected when not initialized', () => {
		jest.spyOn( hooks, 'useRewindState' ).mockReturnValue( { state: 'uninitialized' } );

		const wrapper = shallow( <SettingsPage /> );

		expect( wrapper ).toMatchSnapshot();
	} );
} );

describe( 'ConnectionStatus', () => {
	test( 'should render as expected', () => {
		const wrapper = shallow( <ConnectionStatus /> );

		expect( wrapper ).toMatchSnapshot();
	} );

	test( 'should render as expected when connected', () => {
		const wrapper = shallow( <ConnectionStatus isConnected /> );

		expect( wrapper ).toMatchSnapshot();
	} );
} );
