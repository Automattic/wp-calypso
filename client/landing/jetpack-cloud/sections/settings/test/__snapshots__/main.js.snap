// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectionStatus should render as expected 1`] = `
<Card
  className="settings_disconnected"
  compact={true}
>
  <img
    alt=""
    className="settings__icon"
    src="server-disconnected.svg"
  />
  <div
    className="settings__details"
  >
    <div
      className="settings__details-head"
    >
       
      Server status: Not connected
       
    </div>
    <div>
      Enter your server credentials to enable one-click restores for Backups. 
      <ExternalLink
        className="settings__link-external"
        href="https://jetpack.com/support/adding-credentials-to-jetpack/"
        icon={true}
        iconSize={18}
        key="interpolation-child-1/.0"
        showIconFirst={false}
      >
        Need help? Find your server credentials
      </ExternalLink>
    </div>
  </div>
</Card>
`;

exports[`ConnectionStatus should render as expected when connected 1`] = `
<Card
  className="settings_connected"
  compact={true}
>
  <img
    alt=""
    className="settings__icon"
    src="server-connected.svg"
  />
  <div
    className="settings__details"
  >
    <div
      className="settings__details-head"
    >
       
      Server status: Connected
       
    </div>
    <div>
      One-click restores are enabled.
    </div>
  </div>
</Card>
`;

exports[`SettingsPage should render as expected 1`] = `
<Main
  className="settings"
>
  <Connect(DocumentHead)
    title="Settings"
  />
  <Connect(MySitesSidebarNavigation) />
  <Connect(QueryRewindState)
    siteId={1}
  />
  <Connect(PageViewTracker)
    path="/settings/:site"
    title="Settings"
  />
  <div
    className="settings__title"
  >
    <h2>
      Server connection details
    </h2>
  </div>
  <ConnectionStatus
    isConnected={true}
  />
  <Localized(FoldableCard)
    className="settings__form-card"
    expanded={false}
    header="Show connection details"
    onClick={[Function]}
  >
    <Connect(ServerCredentialsForm)
      labels={
        Object {
          "save": "Save credentials",
        }
      }
      role="main"
      showCancelButton={false}
      siteId={1}
    />
  </Localized(FoldableCard)>
</Main>
`;

exports[`SettingsPage should render as expected when not connected 1`] = `
<Main
  className="settings"
>
  <Connect(DocumentHead)
    title="Settings"
  />
  <Connect(MySitesSidebarNavigation) />
  <Connect(QueryRewindState)
    siteId={1}
  />
  <Connect(PageViewTracker)
    path="/settings/:site"
    title="Settings"
  />
  <div
    className="settings__title"
  >
    <h2>
      Server connection details
    </h2>
  </div>
  <ConnectionStatus
    isConnected={false}
  />
  <Localized(FoldableCard)
    className="settings__form-card"
    expanded={false}
    header="Show connection details"
    onClick={[Function]}
  >
    <Connect(ServerCredentialsForm)
      labels={
        Object {
          "save": "Save credentials",
        }
      }
      role="main"
      showCancelButton={false}
      siteId={1}
    />
  </Localized(FoldableCard)>
</Main>
`;

exports[`SettingsPage should render as expected when not initialized 1`] = `
<Main
  className="settings"
>
  <Connect(DocumentHead)
    title="Settings"
  />
  <Connect(MySitesSidebarNavigation) />
  <Connect(QueryRewindState)
    siteId={1}
  />
  <Connect(PageViewTracker)
    path="/settings/:site"
    title="Settings"
  />
  <div
    className="settings__title"
  >
    <h2>
      Server connection details
    </h2>
  </div>
  <div
    className="settings__status-uninitialized"
  />
  <div
    className="settings__form-uninitialized"
  />
</Main>
`;
